pkg/commands/workflows/plan_status_comments.go

```diff
@@ -132,95 +109,68 @@ func (c *PlanStatusCommentCommand) Run(ctx context.Context, args []string) error
 		return flag.ErrHelp
 	}
 
-	c.actions = githubactions.New(githubactions.WithWriter(c.Stdout()))
-	actionsCtx, err := c.actions.Context()
+	rc, err := c.createReporterClient(ctx)
 	if err != nil {
-		return fmt.Errorf("failed to load github context: %w", err)
+		return fmt.Errorf("failed to create repoter client: %w", err)
 	}
+	c.reporterClient = rc
+
+	return c.Process(ctx)
+}
 
-	c.cfg = &PlanStatusCommentsConfig{}
-	if err := c.cfg.MapGitHubContext(actionsCtx); err != nil {
-		return fmt.Errorf("failed to load github context: %w", err)
+// createReporterClient creates a new reporter client based on the reporter flag.
+func (c *PlanStatusCommentCommand) createReporterClient(ctx context.Context) (reporter.Reporter, error) {
```

Comment:
I dont like this, it needs to be repeated everywhere a reporter needs to be created for a command, which is apply, plan, plan_status_comments, remove_comments, entrypoints